pages/dashboard/index.tsx

```diff
@@ -42,6 +43,7 @@ export default function Dashboard(props: {
   const [challenges, setChallenges] = useState<Challenge[]>([]);
 
   useEffect(() => {
+    firebase.analytics().logEvent('access_dashboard');
```

Comment:
Can we move any calls to `firebase.analytics` to separate `useEffect` hooks? Since it is completely separate from any of the other logic, we should separate it [as suggested by React](https://legacy.reactjs.org/docs/hooks-effect.html#tip-use-multiple-effects-to-separate-concerns).